mlflow/server/js/src/common/utils/AuthComponent.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Utils from "../../common/utils/Utils";
+import { Redirect } from 'react-router';
+import PropTypes from 'prop-types';
+
+class AuthComponentImpl extends Component {
+
+  constructor(props) {
+    super(props);
+    const urlState = Utils.getSearchParamsFromUrl(props.location.search);
+    this.state = {
+      persistedState: {
+        code: urlState.code === undefined ? "" : urlState.code,
+      },
+    };
+  }
+
+  static propTypes = {
+    location: PropTypes.object,
+  };
+
+  componentDidMount = () => {
```

Comment:
Could you add test for this ? To check how to mock XMLhttprequest. (can be injected and mocked using jest ?)
I imagine a test, to check the token is in the localstorage after the component is mounted.

Reply:
Done